### PR TITLE
Make release process slightly better documented

### PR DESCRIPTION
**Pull request by kingdonb**
Tl;dr: it's not inconsistent if the exception is documented.

For all future VS Code Exception developers, here is the release workflow "squash merge" exception

All PRs are expected to be "squashed when merged" except for the `release-pr`. You will only need to know this information if you are a maintainer merging feature contributions, or if you are actually the one physically doing releases.

* If you are merging a feature branch, it is a "**Squash and merge**"
* If you are merging the `release-pr` branch, "**Create a merge commit**"

This unfortunately erases some history inside of a feature branch.

I do not have a good solution for that issue, but I hope it will not be too onerous if we keep feature branches small.

It was necessary to make this compromise in order to achieve a fully auto-generated `CHANGELOG` together with a release machine that handles the version incrementing on our behalf. Release engineers will only need to know if a release includes any breaking change or new feature in the public API to determine whether it should be a MAJOR, MINOR or PATCH release.

### CODE REVIEW

**Comment by kingdonb** [comment]
One positive side-benefit of this policy will be, we never have to ask contributors to rebase their PRs before merging them, unless they really do have an actual merge conflict.

**Review by kingdonb — COMMENTED** [review]
👍
